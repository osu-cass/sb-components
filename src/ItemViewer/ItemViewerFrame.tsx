import * as React from "react";
import { LoadingOverlay } from "../Layout/LoadingOverlay";

export interface FrameProps {
  url: string;
}

export interface FrameState {
  loading: boolean;
}

export class ItemViewerFrame extends React.Component<FrameProps, FrameState> {
  constructor(props: FrameProps) {
    super(props);
    this.state = { loading: true };
  }

  startLoad = () => {
    this.setState({
      loading: true
    });
  };

  finishLoad = () => {
    this.setState({
      loading: false
    });
  };

  renderNoItem() {
    return (
      <div className="no-item">
        <p>No Item Found</p>
      </div>
    );
  }

  componentWillReceiveProps(nextProps: FrameProps) {
    if (nextProps.url !== this.props.url) {
      this.startLoad();
    }
  }

  renderProgressBar() {
    return this.state.loading ? (
      <div className="loader-icon">
        <div className="loader" />
      </div>
    ) : (
      undefined
    );
  }

  renderItem() {
    return (
      <div className="itemViewerFrame" tabIndex={0}>
        {this.renderProgressBar()}
        <iframe
          id="itemviewer-iframe"
          className="itemviewer-iframe"
          onLoadStart={this.startLoad}
          onLoad={this.finishLoad}
          src={this.props.url}
        />
      </div>
    );
  }

  render() {
    if (this.props.url) {
      return this.renderItem();
    } else {
      return this.renderNoItem();
    }
  }
}
