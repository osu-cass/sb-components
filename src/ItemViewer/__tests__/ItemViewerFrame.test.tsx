import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemFrame } from '../ItemViewerFrame';
import * as TestUtils from 'react-dom/test-utils';
import { shallow, mount, render } from 'enzyme';

describe("ItemViewerFrame", () => {
    const obj = {
        url: "http://test.com",
    }

    it("matches snapshot while loading", () => {
        let wrapper = shallow(<ItemFrame {...obj}/>);
        wrapper.setState({loading: true});
        expect(wrapper).toMatchSnapshot();
    })

    it("after page content is loaded", () => {
        let wrapper = shallow(<ItemFrame {...obj}/>);
        wrapper.setState({loading: false});
        expect(wrapper).toMatchSnapshot();
    })

    it("no item found", () => {
        expect(shallow(<ItemFrame url=""/>)).toMatchSnapshot();
    })
})