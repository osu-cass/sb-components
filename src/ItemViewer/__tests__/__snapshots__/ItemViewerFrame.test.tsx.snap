// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemViewerFrame after page content is loaded 1`] = `
<div
  className="itemViewerFrame"
  tabIndex={0}
>
  <iframe
    className="itemviewer-iframe"
    id="itemviewer-iframe"
    onLoad={[Function]}
    onLoadStart={[Function]}
    src="http://test.com"
  />
</div>
`;

exports[`ItemViewerFrame matches snapshot while loading 1`] = `
<div
  className="itemViewerFrame"
  tabIndex={0}
>
  <div
    className="loader-icon"
  >
    <div
      className="loader"
    />
  </div>
  <iframe
    className="itemviewer-iframe"
    id="itemviewer-iframe"
    onLoad={[Function]}
    onLoadStart={[Function]}
    src="http://test.com"
  />
</div>
`;

exports[`ItemViewerFrame no item found 1`] = `
<div
  className="no-item"
>
  <p>
    No Item Found
  </p>
</div>
`;
